apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx

```diff
@@ -273,23 +453,88 @@ function ArtifactSelectModal({
               filteredIds={artIdList}
             />
           </Suspense>
+        </CardContent>
+        <Divider />
+        <CardContent>
           <Box mt={1}>
             <Suspense
               fallback={
                 <Skeleton variant="rectangular" width="100%" height={300} />
               }
             >
-              <Grid container spacing={1} columns={{ xs: 2, md: 3, lg: 4 }}>
-                {artIdList.map((id) => (
-                  <Grid item key={id} xs={1}>
-                    <ArtifactCard artifactId={id} onClick={clickHandler} />
+              <Grid container alignItems="center" sx={{ pb: 1 }}>
+                <Grid item flexGrow={1}>
+                  <Pagination
+                    count={numPages}
+                    page={currentPageIndex + 1}
+                    onChange={setPage}
+                  />
+                </Grid>
+                <Grid item flexGrow={1}>
+                  <ShowingArt
+                    numShowing={artifactIdsToShow.length}
+                    total={totalShowing}
+                    t={tk}
+                  />
+                </Grid>
+              </Grid>
+              <ExcludeArtRedButtons artifactIds={artIdList} />
+              <Grid
+                container
+                spacing={1}
+                columns={{ xs: 2, md: 3, lg: 4 }}
+                sx={{ pt: 1 }}
+              >
+                {artifactIdsToShow.map((id) => (
+                  <Grid
+                    item
+                    key={id}
+                    xs={1}
+                    sx={{ ...(artExclusion.includes(id) && { opacity: 0.4 }) }}
+                  >
+                    <ArtifactCard
+                      artifactId={id}
+                      onClick={clickHandler}
```

Comment:
We should change this clickHandler to now include artifacts, if you click an excluded one

Reply:
Implemented